### PR TITLE
fix(routing): fix routing with url encoded values

### LINKED ISSUE #517: unexpected 404
I got a 404 after clicking around for a while: https://ops.grafana-ops.net/a/grafana-lokiexplore-app/explore/service/Hosted%20Grafana%20-%20Prod/logs?patterns=%5B%5D&var-fields=&var-ds=grafanacloud-logs&var-patterns=&var-lineFilter=&var-logsFormat=&var-filters=service_name%7C%3D%7CHosted%20Grafana%20-%20Prod

I notice filters were persisted in the URL as I selected the service.

### PR DESCRIPTION
**Pull request by svennergr**
Links with URL encoded values for e.g. `:service` would get double URL encoded and thus don't match any of our defined routes.

This PR just removes the unnecessary additionally routing.

Fixes #517

### CODE REVIEW

**Comment by svennergr** [comment]
Something we might want to do before merging is to test this with a service name containing a slash. Whitespaces worked fine with this fix.

**Comment by matyax** [comment]
Yeah, it's not working with services with slashes.

https://github.com/grafana/explore-logs/assets/1069378/715ca065-4a63-40bb-a6ab-e59a2761857d

**Comment by svennergr** [comment]
> Yeah, it's not working with services with slashes.
> 
>  Nope.mov

Mind testing again? Thanks!

File: src/services/routing.ts [hunk 1]
```diff
@@ -24,23 +24,16 @@ export enum PageSlugs {
...
+    prefixRoute(`${PageSlugs.explore}/service/${service.replace(/\//g, '-')}/${PageSlugs.logs}`),
   fields: (service: string) =>
-    prefixRoute(`${PageSlugs.explore}/service/${encodeParameter(service)}/${PageSlugs.fields}`),
+    prefixRoute(`${PageSlugs.explore}/service/${service.replace(/\//g, '-')}/${PageSlugs.fields}`),
   patterns: (service: string) =>
-    prefixRoute(`${PageSlugs.explore}/service/${encodeParameter(service)}/${PageSlugs.patterns}`),
+    prefixRoute(`${PageSlugs.explore}/service/${service.replace(/\//g, '-')}/${PageSlugs.patterns}`),
   labels: (service: string) =>
-    prefixRoute(`${PageSlugs.explore}/service/${encodeParameter(service)}/${PageSlugs.labels}`),
+    prefixRoute(`${PageSlugs.explore}/service/${service.replace(/\//g, '-')}/${PageSlugs.labels}`),
```
Comment (matyax) [inline src/services/routing.ts:36]: I just realized now, but we could have implemented the replace within `encodeParameter()` so you don't need to `service.replace()` every route.

**Review by matyax — APPROVED** [review]
It fixes the problem!

**Pull request merged**